### DISCUSSION TITLE
How could i create growable DataTable dynamicly？

### DISCUSSION THREAD

**Discussion by zhajio1988** [Textualize/textual#2495] [Q&A] [answered]
I found textual has a widget DataTable， but it's static， how could i create growable DataTable dynamicly？

**Comment by davep** [comment]
What do you man when you say it's "static"? Have you had a look at [the documentation for `DataTable`](https://textual.textualize.io/widgets/data_table/), it talks about methods for adding and removing data, amongst other things.

**Reply by zhajio1988** [reply]
@davep 
Hi, Dave. I want to read a table from the database and fill it into the DataTable, but the database is updated in real time by another script, so I want to implement the DataTable to update every few seconds and append new rows to the end of the table.

**Reply by davep** [reply, marked as answer]
In that case I imagine the approach to take would be to look for changes to your database somehow, or be informed of changes, and when you know there are some load up the new data and [add the new rows](https://textual.textualize.io/widgets/data_table/#textual.widgets._data_table.DataTable.add_row). Exactly how you go about that will depend heavily on the nature of your application.

Things that might be helpful here are [`set_interval`](https://textual.textualize.io/api/message_pump/#textual.message_pump.MessagePump.set_interval) or perhaps [workers](https://textual.textualize.io/guide/workers/).

**Reply by zhajio1988** [reply]
@davep 

Thanks, dave. set_interval maybe work for my APP.

**Reply by zhajio1988** [reply]
@davep 
Hi Dave, 
```
class TableApp(App):
    def compose(self) -> ComposeResult:
        yield DataTable()

    def on_mount(self) -> None:
        table = self.query_one(DataTable)
        self.rowIdx = 0
        table.add_columns(*ROWS[0])
        table.add_rows(ROWS[1:])
        self.update_timer = self.set_interval(2, self.update_time)

    def update_time(self) -> None:
        """Method to update time to current."""
        table = self.query_one(DataTable)
        table.add_rows(ROWS[2:3])
        table.focus()

app = TableApp()
if __name__ == "__main__":
    app.run()
```
Above code can add row dynamically,  but during this stage, CTRL+C can't exit app, why this happens？

**Reply by davep** [reply]
Sounds like it might be related to #2492.
